src/firstboot/firstboot.c

```diff
@@ -802,7 +802,7 @@ static int write_root_shadow(const char *shadow_path, const char *hashed_passwor
 
 static int process_root_args(void) {
         _cleanup_close_ int lock = -1;
-        struct crypt_data cd = {};
+        _cleanup_free_ char *_hashed_password = NULL;
```

Comment:
`_cleanup_(erase_and_freep)_` is probably better

Reply:
Yeah, I should have thought of that myself.